Client/src/Components/Graph/Graph.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component } from 'react'
+import * as d3 from "d3"
+
+//Global variables for graph creation
+
+var margin = {
+    top: 50,
+    right: 210,
+    bottom: 50,
+    left: 70
+},
+    outerWidth = 1050,
+    outerHeight = 500,
+    width = outerWidth - margin.left - margin.right,
+    height = outerHeight - margin.top - margin.bottom;
+
+var xScale, yScale, xAxis, yAxis;
+
+//This function creates the basis for the linear x and y axis
+function startLinear() {
+    xScale = d3.scaleLinear()
+        .range([0, width]).nice();
+
+    yScale = d3.scaleLinear()
+        .range([height, 0]).nice();
+
+    xScale.domain([0, 10]);
+    yScale.domain([0, 10]);
+
+    xAxis = d3.axisBottom()
+        .scale(xScale)
+        .tickSize(-height);
+
+    yAxis = d3.axisLeft()
+        .scale(yScale)
+        .tickSize(-width);
+
+}
+
+//sample datasets to try, just needs to gather from the backend instead.
+var dataset1 = [{ x: 5.1, y: 3.5 }, { x: 4.9, y: 3 }, { x: 4.7, y: 3.2 }, { x: 4.6, y: 3.1 }, { x: 5, y: 3.6 }, { x: 5.4, y: 3.9 }]
+var dataset2 = [{ x: 3, y: 8 }, { x: 9, y: 7 }, { x: 1, y: 3 }, { x: 2, y: 4 }, { x: 8, y: 1 }]
+var dataset3 = [{ x: 7, y: 6 }, { x: 2, y: 5 }, { x: 7, y: 9 }, { x: 4, y: 1 }, { x: 6, y: 7 }]
+var dataset4 = [{ x: 3, y: 7 }, { x: 1, y: 9 }, { x: 8, y: 7 }, { x: 1, y: 4 }, { x: 8, y: 5 }]
+//Datalist is the list fed to the graphCreation
+var datalist = []
+var colourslist = []
+datalist.push(dataset1)
+datalist.push(dataset2)
+datalist.push(dataset3)
+datalist.push(dataset4)
+
+class Graph extends Component {
```

Comment:
Class components are also old-age... Functional components all the way. This lets us use Hooks

Reply:
Working on it, will create another issue for this problem